gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/services/RemoveRemoteVoltageRegulationInFranceService.java

```diff
@@ -30,23 +32,38 @@ public RemoveRemoteVoltageRegulationInFranceService(DataFixConfiguration dataFix
         this.businessLogger = businessLogger;
     }
 
-    public void removeRemoteVoltageRegulationInFrance(Network network) {
+    public Map<String, ReplacedVoltageRegulation> removeRemoteVoltageRegulationInFrance(Network network) {
         if (!dataFixConfiguration.removeRemoteVoltageRegulationInFrance()) {
-            return;
+            return Collections.emptyMap();
         }
         businessLogger.info("Removing remote voltage regulations in France.");
-        network.getGeneratorStream()
+        return network.getGeneratorStream()
                 .filter(RemoveRemoteVoltageRegulationInFranceService::isRemoteVoltageRegulationOn)
                 .filter(RemoveRemoteVoltageRegulationInFranceService::isInFrance)
-                .forEach(this::removeRemoteVoltageRegulation);
+                .collect(Collectors.toMap(
+                        Generator::getId,
+                        this::removeRemoteVoltageRegulation
+                ));
+    }
+
+    public void resetRemoteVoltageRegulationInFrance(Network network, Map<String, ReplacedVoltageRegulation> generatorsExtension) {
+        generatorsExtension.forEach((generatorId, replacedVoltageRegulation) -> {
+            Generator generator = network.getGenerator(generatorId);
+            Terminal regulatingTerminal = network.getConnectable(replacedVoltageRegulation.regulatingTerminalConnectableId).getTerminals().stream().filter(terminal -> terminal.getSide() == replacedVoltageRegulation.regulatingTerminalConnectableSide).findAny().get();
```

Comment:
Perhaps replace the .get by .orElseThrow with a SweInvalidDataException indicating the missing terminal for instance, so that logs will be easier to understand if such a case happens